Create Grants page in Community section

jhaaaa opened the pull request:
Preview here: https://junk-range-possible-git-xmtp-grants-xmtp-labs.vercel.app/grants

**Open questions:**

- What do you think of the order of pages in the Community drop-down?
- What do you think of using the x-mark icon? If we want a rocket - we can ask Darick to create the icon for us.
- In Docusaurus, we can't create the two-column layout in the Request for proposals section using a humble markdown file. <3 I don't think it's a dealbreaker, but defer to you, of course!
- What do you think of the expand/collapse modules? "Click for details" is customizable - so we can use different CTA text, if you'd like.
- Need XMTP brand assets link before we go live

yash-luna commented:
Looks awesome! Some smol suggestions & follow-ups:

- Should we push Grants to #2 in the dropdown since we want more attention at launch?
- Will ask Darick if we can get a rocket icon
- Would the page look better without the collapse modules (since we don't have them anywhere else on our website)?
- Might have to remove that brand assets link for now because the Design repo isn't ready
- Can we get callout boxes for 'Apply here' to drive attention to CTA? We'll have to differentiate them from the 'Note' callout. Docusaurus has some nice options - https://docusaurus.io/docs/markdown-features/admonitions

jhaaaa commented:
Per @yash-luna:

- Updated page to match Notion content
- Moved Grants to position 2 (after join) in drop-down and left nav
- Used rocket icon from Darick in drop-down
- Removed collapse modules
- Removed brand assets link for now
- Used callout boxes for 'Apply here'
- Moved Resources section to end right before Contact us
